be/src/vec/exec/format/parquet/vparquet_reader.h

```diff
@@ -48,40 +44,46 @@ namespace doris::vectorized {
 
 class ParquetReader {
 public:
-    ParquetReader(FileReader* file_reader, int64_t batch_size, int32_t num_of_columns_from_file,
+    ParquetReader(FileReader* file_reader, int32_t num_of_columns_from_file,
                   int64_t range_start_offset, int64_t range_size);
+
     ~ParquetReader();
-    virtual Status init_reader(const TupleDescriptor* tuple_desc,
-                               const std::vector<SlotDescriptor*>& tuple_slot_descs,
-                               const std::vector<ExprContext*>& conjunct_ctxs,
-                               const std::string& timezone) = 0;
-    virtual Status next_batch(bool* eof) = 0;
+
+    Status init_reader(const TupleDescriptor* tuple_desc,
+                       const std::vector<SlotDescriptor*>& tuple_slot_descs,
+                       const std::vector<ExprContext*>& conjunct_ctxs, const std::string& timezone);
+
+    Status read_next_batch(Block* block);
+
+    bool has_next() const { return !_batch_eof; };
+
     //        std::shared_ptr<Statistics>& statistics() { return _statistics; }
-    void close() {};
-    int64_t size(int64_t* size) { return _file_reader->size(); }
+    void close();
+
+    int64_t size() const { return _file_reader->size(); }
 
 private:
-    int64_t _get_row_group_start_offset(const tparquet::RowGroup& row_group);
+    Status _column_indices(const std::vector<SlotDescriptor*>& tuple_slot_descs);
+    void _init_row_group_reader();
+    void _fill_block_data(std::vector<tparquet::ColumnChunk> columns);
+    bool _has_page_index(std::vector<tparquet::ColumnChunk> columns);
+    Status _process_page_index(std::vector<tparquet::ColumnChunk> columns);
 
 private:
     FileReader* _file_reader;
     std::shared_ptr<FileMetaData> _file_metadata;
-    //    const int64_t _batch_size;
-    //    const int32_t _num_of_columns_from_file;
+    std::shared_ptr<RowGroupReader> _row_group_reader;
```

Comment:
In the new design, we should ensure that the row group reader is a reader that can be processed independently in parallel.

Reply:
stay implement after this version